test/functional/repository/find-options/repository-find-options.ts

```diff
@@ -199,16 +199,16 @@ describe("repository > find options > cache", () => {
         user4.name = "Ginny";
         await connection.manager.save(user4);
 
-        // without cache it must return really how many there entities are
-        const users2 = await connection.getRepository(User).find();
-
-        expect(users2.length).to.be.equal(4);
-
-        // but with cache enabled it must not return newly inserted entity since cache is not expired yet
+        // with the cache enabled (and not expired) we expect our previously cached results i.e. not the newly inserted entity
```

Comment:
I guess I need explanation on changes in every test. Why it didn't work before, and how new changes fixed it

Reply:
The reasoning is covered in the commit message for this change. Basically there's a potential timing issue any time the cache is involved as the cache expires after 1 second.

Previously the test went:

1. Insert
2. Verify we can retrieve from DB
3. Verify we can retrieve from cache

If step 2 takes too long the cache expires by the time we reach 3 (in practice this was occurring for me sporadically when running the tests in Docker on macOS i.e. a VM). So I've reordered the test to:

1. Insert
2. Verify we can retrieve from cache
3. Verify we can retrieve from DB

Whilst in _theory_ the cache can still expire, it's now exceedingly unlikely as we're using the cache immediately.